File: wf-beneficiary-management/template/java/model/exception/WfException.java
package {basePackage}.wf.model.exception;

import org.apache.commons.lang3.builder.ToStringBuilder;
import org.apache.commons.lang3.builder.ToStringStyle;
import lombok.Getter;

/**
 * WorldFirst API exception.
 */
@Getter
public class WfException extends RuntimeException {

    private final WfErrorCode errorCode;

    public WfException(WfErrorCode errorCode, String message) {
        super(message);
        this.errorCode = errorCode;
    }

    public WfException(WfErrorCode errorCode, String message, Throwable cause) {
        super(message, cause);
        this.errorCode = errorCode;
    }

    @Override
    public String toString() {
        return ToStringBuilder.reflectionToString(this, ToStringStyle.SHORT_PREFIX_STYLE);
    }
}

File: new-skills/wf-api-integration/references/common/golang/model/exception/wf_exception.go
package exception

import "fmt"

// WfException represents a WorldFirst API exception
type WfException struct {
	Code    WfErrorCode
	Message string
}

func (e *WfException) Error() string {
	return fmt.Sprintf("WF Error [%s]: %s", e.Code, e.Message)
}

// NewWfException creates a new WfException
func NewWfException(code WfErrorCode, message string) *WfException {
	return &WfException{Code: code, Message: message}
}

